drivers/input/input_tma525b.c

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright (c) 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT parade_tma525b
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/input/input_touch.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(tma525b, CONFIG_INPUT_LOG_LEVEL);
+
+/* TMA525B I2C address */
+#define TMA525B_I2C_ADDRESS 0x24U
+
+/* TMA525B maximum number of simultaneously detected touches */
+#define TMA525B_MAX_TOUCHES 4U
+
+/* TMA525B register address where touch data begin */
+#define TMA525B_TOUCH_DATA_SUBADDR 1U
+
+/* TMA525B raw touch data length */
+#define TMA525B_TOUCH_DATA_LEN 264U
+
+/* TMA525B touch data header length to read first */
+#define TMA525B_TOUCH_DATA_LEN_BYTES 2U
+
+/* TMA525B report ID for touch data */
+#define TMA525B_REPORT_ID_TOUCH 0x01U
+
+/* Touch event types - matching fsl_tma525b.h */
+enum touch_event {
+	TOUCH_RESERVED = 0, /* No touch event detected */
+	TOUCH_DOWN = 1,     /* Touch down event detected */
+	TOUCH_CONTACT = 2,  /* Touch point moving */
+	TOUCH_UP = 3        /* Touch event finished */
+};
+
+/* TMA525B touch point structure */
+struct tma525b_touch_point {
+	uint16_t touch_type : 8U; /* 0 for standard finger/glove, 1 for proximity. Not used*/
+	uint16_t event_id : 8U;   /* Bit 0-4: touch ID, bit 5-6: touch event */
+	uint16_t x;
+	uint16_t y;
+	uint16_t pressure : 8U;       /* Touch intensity. Not used in current driver model. */
+	uint16_t major_axis_len_mm : 8U; /* Major axis length. Not used in current driver model. */
+	uint16_t minor_axis_len_mm : 8U; /* Minor axis length. Not used in current driver model. */
+	/* Angle between panel vertical axis and major axis. Not used in current driver model. */
+	uint16_t orientation : 8U;
+} __packed;
+
+/* TMA525B touch data structure */
+struct tma525b_touch_data {
+	uint16_t length; /* Packet length. */
+	uint8_t report_id;
+	/* Timestamp in 100us units low byte. Not used in current driver model. */
+	uint8_t timestamp_100us_l;
+	/* Timestamp in 100us units high byte. Not used in current driver model. */
+	uint8_t timestamp_100us_h;
+	uint8_t num_touch; /* Number of touch points detected */
+	/* 2 MSB: report counter, 3 LSB: noise effects. Not used in current driver model. */
+	uint8_t report_noise;
+	struct tma525b_touch_point touch[TMA525B_MAX_TOUCHES];
+} __packed;
+
+/* Macros for extracting touch data. */
+#define TMA525B_TOUCH_POINT_GET_ID(event_id)    (event_id & 0x1F)
+#define TMA525B_TOUCH_POINT_GET_EVENT(event_id) (((event_id) & 0x60) >> 5U)
```

Comment:
drop the parenthesis from around `event_id`? you don't have them in the line above anyway